packages/atomic/src/components/common/tabs/tab-bar.tsx

```diff
@@ -0,0 +1,186 @@
+import {h, Component, Element, Host, Listen, State} from '@stencil/core';
+import {Button} from '../button';
+import {TabCommonElement} from './tab-common';
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'tab-bar',
+  shadow: true,
+  styleUrl: 'tab-bar.pcss',
+})
+export class TabBar {
+  @Element() private host!: HTMLElement;
+
+  @State()
+  private popoverTabs: typeof Button[] = [];
+
+  private get tabsFromSlot(): TabCommonElement[] {
+    const isTab = (tagName: string) => /atomic-.+-tab$/i.test(tagName);
+    return Array.from(this.host.querySelectorAll<TabCommonElement>('*')).filter(
+      (element) => isTab(element.tagName)
+    );
+  }
+
+  private get selectedTab() {
+    return this.tabsFromSlot.find((tab) => tab.active);
+  }
+
+  private get slotContentWidth() {
+    return this.tabsFromSlot.reduce(
+      (total, el) =>
+        total +
+        parseFloat(window.getComputedStyle(el).getPropertyValue('width')),
+      0
+    );
+  }
+
+  private get containerWidth() {
+    return parseFloat(
+      window.getComputedStyle(this.host).getPropertyValue('width')
+    );
+  }
+
+  private get isOverflow() {
+    return this.slotContentWidth > this.containerWidth;
+  }
+
+  private get tabPopover() {
+    return this.host.shadowRoot?.querySelector('tab-popover');
+  }
+
+  private get popoverWidth() {
+    return this.tabPopover ? this.getElementWidth(this.tabPopover) : 0;
+  }
+
+  private get overflowingTabs() {
+    const containerRelativeRightPosition =
+      this.host.getBoundingClientRect().right;
+    const selectedTabRelativeRightPosition =
+      this.selectedTab?.getBoundingClientRect().right;
+
+    return this.tabsFromSlot.filter((element) => {
+      const isBeforeSelectedTab = selectedTabRelativeRightPosition
+        ? selectedTabRelativeRightPosition >
+          element.getBoundingClientRect().right
+        : null;
+
+      const minimumWidthNeeded = isBeforeSelectedTab
+        ? this.popoverWidth + this.getElementWidth(this.selectedTab)
+        : this.popoverWidth;
+
+      const rightPositionLimit = !this.isOverflow
+        ? containerRelativeRightPosition
+        : containerRelativeRightPosition - minimumWidthNeeded;
+
+      return (
+        element.getBoundingClientRect().right > rightPositionLimit &&
+        !element.active
+      );
+    });
+  }
+
+  private get displayedTabs() {
+    return this.tabsFromSlot.filter((el) => !this.overflowingTabs.includes(el));
+  }
+
+  private get lastDisplayedTab() {
+    const displayedTabs = this.displayedTabs;
+    return displayedTabs[displayedTabs.length - 1];
```

Comment:
Just found out the [at function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at) does handle negative indices.

```suggestion
    return this.displayedTabs.at(-1);
```

Reply:
TIL. It's a good suggestion but unfortunately it doesn't look like the project's target version 😅 

<img width="962" alt="Screen Shot 2023-04-13 at 1 26 21 PM" src="https://user-images.githubusercontent.com/16785453/231850240-23ff17ad-140b-4e3f-b10f-75c6150f9162.png">

Reply:
Ah... sorry :sweat_smile: